language/dsl/python/src/navigation.py

```diff
@@ -0,0 +1,377 @@
+
+# Navigation related classes
+from typing import Any, Dict, Generic, List, Literal, Optional, TypeVar, Union
+
+from data import Expression, ExpressionObject
+
+
+# Type variables for generic classes
+T = TypeVar('T', bound=str)
+
+class Navigation:
+    """The navigation section of the flow describes a State Machine for the user."""
+    
+    def __init__(self, begin: str, **flows: Union[str, 'NavigationFlow']):
+        self._begin = begin
+        self._flows: Dict[str, Union[str, 'NavigationFlow']] = flows
+    
+    @property
+    def begin(self) -> str:
+        """The name of the Flow to begin on"""
+        return self._begin
+    
+    @begin.setter
+    def begin(self, value: str) -> None:
+        self._begin = value
+    
+    def get_flow(self, name: str) -> Optional[Union[str, 'NavigationFlow']]:
+        """Get a flow by name"""
+        return self._flows.get(name)
+    
+    def set_flow(self, name: str, flow: Union[str, 'NavigationFlow']) -> None:
+        """Set a flow"""
+        self._flows[name] = flow
+    
+    @property
+    def flows(self) -> Dict[str, Union[str, 'NavigationFlow']]:
+        """Get all flows"""
+        return self._flows.copy()
+
+
+NavigationFlowTransition = Dict[str, str]
+
+
+class CommentBase:
```

Comment:
Overcomplicating things again, could the name that it serializes to be configurable? I've seen folks put `_comment` in their JSON rather than `comment`. Unsure if that'd be something that is even relevant to this use case.

Reply:
This just got ported over for completeness. Since folks shouldn't be writing JSON directly is it something we should still account for?

Reply:
After offline discussion this will be omitted.